src/fides/ctl/core/dataset.py

```diff
@@ -319,4 +344,7 @@ def generate_bigquery_datasets(bigquery_config: BigQueryConfig) -> List[Dataset]
     bigquery_engine = get_bigquery_engine(bigquery_config)
     bigquery_schemas = get_db_schemas(engine=bigquery_engine)
     bigquery_datasets = create_db_datasets(db_schemas=bigquery_schemas)
-    return bigquery_datasets
+    unique_bigquery_datasets = make_dataset_fides_key_unique(
```

Comment:
This is a design comment/nit, but generally I think it is better to loop _outside_ of the lower-level functions and instead do it in the main codepath.

As an example, this is how I would rewrite this line

`unique_bigquery_datasets = [make_dataset_key_unique(dataset) for dataset in bigquery_datasets]`

The above function would then be updated to only take a single dataset instead of a list of them.

Does that make sense/do you agree with that opinion?

Reply:
Thank you for taking the time to lay that out! I am definitely ok with that. I think it is sort of an existing pattern currently elsewhere but am happy to start making an effort to standardize around this 👍🏽